esp8266-weather-station-color.ino

```diff
@@ -238,15 +260,38 @@ uint8_t currentTouchPoint = 0;
 void loop() {
   static bool asleep = false; //  asleep used to stop screen change after touch for wake-up
   gfx.fillBuffer(MINI_BLACK);
-  if (touchController.isTouched(0)) {
+
+  /* Break up the screen into 4 sections a touch in section:
+   * - Top changes the time format
+   * - Left back one page
+   * - Right forward one page
+   * - Bottom jump to page 0
+   */
+  const static uint8_t top = gfx.getHeight() / 4;
+  const static uint8_t bottom = gfx.getHeight() - (gfx.getHeight() / 4);
+  const static uint8_t middle = gfx.getWidth() / 2;
+  if (touchController.isTouched(500)) {
     TS_Point p = touchController.getPoint();
     timerPress = millis();
-
+    
     Serial.printf("Touch point detected at %d/%d.\n", p.x, p.y);
-    if (!asleep) { // no need to change screens;
-      if (p.y < 80) {
+    if (!asleep) { // no need to update or change screens;
+      if (p.y <= top)   Serial.print(" TOP ");
```

Comment:
Since these are not if-else-if you might see "Bottom Right" on the console even though no action is associated with that specific section (only "Bottom" is relevant in that context). That made me wonder...

Ideally 
- all of that logic should go into a "ScreenSection"-like class or function
- the evaluation of which section was clicked should accept a `TS_Point` and return a section enum: TOP, BOTTOM, MIDDLE_LEFT, MIDDLE_RIGHT

Reply:
- That is by design for the debugging output the extra space in the string gives output like " Top Left " or " Right Bottom " the screen selection part is using the nested if then else.
- Agree it would be better to do the screen selection and draw screen in a function call outside of loop().
- using enum would allow using case instead of if/then/else.
I'll work on making these changes, thanks.

Reply:
Any thing else I need to do?